File: LeetCode/451. Sort Characters By Frequency.py
### KNOWLEDGE IS POWER  ### 

class Solution(object):
    def frequencySort(self, s):
        """
        :type s: str
        :rtype: str
        """
        result = ''
        bucket = [None for i in range(len(s) + 1)]
        hash_map = {}
        
        for char in s:
            hash_map[char] = hash_map.get(char, 0) + 1
            
        for key, value in hash_map.items():
            if bucket[value] is None:
                bucket[value] = []
            
            bucket[value].append(key)
            
        for i in reversed(range(len(bucket))):
            if bucket[i] is not None:
                for char in bucket[i]:
                    result += char * i
                    
        return result

File: LeetCode/451. Sort Characters By Frequency.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    string frequencySort(string s) {
        vector<pair<int,int>> freq(256);
        for(auto c:s) {
            freq[c].first++;
            freq[c].second = c;
        }
        sort(freq.begin(),freq.end(), greater<pair<int,int>> ());
        string ans;
        for(int i = 0; i < 256; i++){
            while(freq[i].first > 0){
                ans += freq[i].second;
                freq[i].first--;
            }
        }
        return ans;
    }
};
